### ISSUE TITLE
Option or command to clear cache (and reload)

### ISSUE THREAD

**Issue by Mikaela** [Nheko-Reborn/nheko#1242]
### The Problem

Often Matrix client and server start disagreeing on the state of the room, so I would like to drop the state on Nheko and have it sync again with the server.

This happens especially when using Nheko together with Dendrite. Many rooms just lose their avatar and name and instead become "someone and 15 others" ("someone and 15 muuta" when using Nheko in Finnish, have we not localised that "and" or is it untranslateable?).

### The Solution

I would like a `/clearcache` command which is familiar from FluffyChat and doesn't require clicking around menus. Having a button in settings like in Element would be ok too, while I would prefer it to be quickly keyboard accessible.

### Alternatives

I don't think there are alternatives.

### Additional context

I apologise if I failed at describing the issue technically, I imagine everyone using Matrix for a while has had experience demanding the use of "clear cache and reload" in Element \<platform\> settings or `/clearcache` in FluffyChat`.

This was previously requested by @Cadair in apparently opposite issue https://github.com/Nheko-Reborn/nheko/issues/360#issuecomment-768657005.

### Happens in the latest version

- [X] Yes, this feature is still missing.

**Comment by deepbluev7** [comment]
This is very unlikely to happen. Please fix the server instead. There is a command to reload the state for a room, /reset-state, but in general I think such buttons are a code smell and you should fix the bugs instead of working around them. I think any client that has them becomes lazy and instead of fixing the bugs, tells their users to press that button instead, and I don't like it.

**Comment by olmari** [comment]
So in the negation of it, just let the user burn in hell no matter where the end flaw is, assuming such clear remedies situation?

**Comment by deepbluev7** [comment]
You can repair a few rooms manually using /reset-state and /clear-timeline, but in general Matrix will never improve, if you just tell people to click a button every time they have an issue instead of fixing the issue properly. I have to click that button every month in Element, because those issues go unfixed and I think that is unacceptable. The button might look like a solution in the short term, but it makes the problems worse in the long term and is very inaccessible.

**Comment by olmari** [comment]
So go figure out what is rootcause in Synapse or whichever server and make PR?

**Comment by deepbluev7** [comment]
Yes, but I hadn't had any such issues on Synapse in years in Nheko (apart from the one time I accidentally reset my database by a few months). It mostly affects dendrite right now, afaik.

**Comment by Mikaela** [comment]
I am told that Dendrite side tracking is either https://github.com/matrix-org/dendrite/issues/2910 or at least similar

**Comment by deepbluev7** [comment]
Afaik dendrite also sometimes sent rooms in the "left" entries, which made Nheko forget about them. Not sure if there is an issue for that or it is fixed or so.

**Comment by Midek** [comment]
There are often glitches like rooms that do not exist anymore or broken unsent messages even when using synapse. In the end lack of option to clear cache just makes the user have to completely logout and log in again instead, and thats not the greatest experience.

**Comment by Mikaela** [comment]
If you only have a few problematic rooms, you can copy-paste `/reset-state` around, but if you have more of the rooms that does get tedious.

**Comment by Midek** [comment]
I cant even /reset-state in the broken rooms because they do not actually exist - i left them some time ago on other client. But nheko would still think im there if i did not relog

**Comment by fgaz** [comment]
This is not only useful for fixing inconsistencies, but also for clearing up space. My `data.mdb` is a few gigabytes

**Comment by jultty** [comment]
For those landing here out of concern with `data.mdb` taking up gigabytes of space, consider starting nheko with `nheko --compact`, which, as per the `--help` output:

```
  -C, --compact            Recompacts the database which might improve
                           performance.
```

This has drastically reduced `data.mdb` space for me, from ~12 GiB to 92 MiB.
